xds/internal/balancer/xdsrouting/matcher.go

```diff
@@ -47,6 +49,14 @@ func (a *compositeMatcher) match(info balancer.PickInfo) bool {
 	var md metadata.MD
 	if info.Ctx != nil {
 		md, _ = metadata.FromOutgoingContext(info.Ctx)
+		if extraMD, ok := grpcutil.ExtraMetadata(info.Ctx); ok {
+			md = metadata.Join(md, extraMD)
+			// Only add grpc-timeout if ctx has extra metadata.
```

Comment:
Why here instead of when setting the metadata in grpc?

Reply:
Removed